x-pack/plugins/fleet/server/services/epm/agent/agent.test.ts

```diff
@@ -363,3 +363,39 @@ paths:
     );
   });
 });
+
+describe('encode', () => {
+  it('should correctly percent encode a string', () => {
+    const streamTemplate = `
+    hosts: 
+      - sqlserver://{{percent_encode username}}:{{percent_encode password}}@{{hosts}}`;
```

Comment:
Thanks for the tests! Can we add some special characters to username too, to verify the helper there.

Reply:
Updated the tests to cover username special characters.